src/riak_kv_qry_compiler.erl

```diff
@@ -113,6 +128,14 @@ prepend_select_columns([_|_] = MultiCols, Acc) ->
 prepend_select_columns(V, Acc) ->
     [V | Acc].
 
+%% copy pasta mapfoldl from lists.erl, otherwise this reports
+%% a warning in dialyzer!
+my_mapfoldl(F, Accu0, [Hd|Tail]) ->
```

Comment:
It is the dialyzer potential bug - I have asked Andrei to create a Jira ticket for this or add to the one Andy created.